command/validate/validate_test.go

```diff
@@ -38,16 +38,62 @@ func TestValidateCommand_SucceedOnMinimalConfigFile(t *testing.T) {
 
 	fp := filepath.Join(td, "config.json")
 	err := ioutil.WriteFile(fp, []byte(`{"bind_addr":"10.0.0.1", "data_dir":"`+td+`"}`), 0644)
-	if err != nil {
-		t.Fatalf("err: %s", err)
-	}
+	require.Nilf(t, err, "err: %s", err)
 
 	cmd := New(cli.NewMockUi())
 	args := []string{fp}
 
-	if code := cmd.Run(args); code != 0 {
-		t.Fatalf("bad: %d", code)
-	}
+	code := cmd.Run(args)
+	require.Equalf(t, 0, code, "bad: %d", code)
+}
+
+func TestValidateCommand_SucceedWithMinimalJSONConfigFormat(t *testing.T) {
+	t.Parallel()
+	td := testutil.TempDir(t, "consul")
+	defer os.RemoveAll(td)
+
+	fp := filepath.Join(td, "json.conf")
+	err := ioutil.WriteFile(fp, []byte(`{"bind_addr":"10.0.0.1", "data_dir":"`+td+`"}`), 0644)
+	require.Nilf(t, err, "err: %s", err)
+
+	cmd := New(cli.NewMockUi())
+	args := []string{"--config-format", "json", fp}
+
+	code := cmd.Run(args)
+	require.Equalf(t, 0, code, "bad: %d", code)
+}
+
+func TestValidateCommand_SucceedWithMinimalHCLConfigFormat(t *testing.T) {
+	t.Parallel()
+	td := testutil.TempDir(t, "consul")
+	defer os.RemoveAll(td)
+
+	fp := filepath.Join(td, "hcl.conf")
+	err := ioutil.WriteFile(fp, []byte("bind_addr = \"10.0.0.1\"\ndata_dir = \""+td+"\""), 0644)
+	require.Nilf(t, err, "err: %s", err)
+
+
+	cmd := New(cli.NewMockUi())
+	args := []string{"--config-format", "hcl", fp}
+
+	code := cmd.Run(args)
+	require.Equalf(t, 0, code, "bad: %d", code)
+}
+
+func TestValidateCommand_SucceedWithJSONAsHCL(t *testing.T) {
+	t.Parallel()
+	td := testutil.TempDir(t, "consul")
+	defer os.RemoveAll(td)
+
+	fp := filepath.Join(td, "json.conf")
+	err := ioutil.WriteFile(fp, []byte(`{"bind_addr":"10.0.0.1", "data_dir":"`+td+`"}`), 0644)
+	require.Nilf(t, err, "err: %s", err)
+
+	cmd := New(cli.NewMockUi())
+	args := []string{"--config-format", "hcl", fp}
+
+	code := cmd.Run(args)
+	require.Equalf(t, 0, code, "bad: %d", code)
```

Comment:
require's error message is pretty good already, so just use `require.Equal(t, expected, actual)` here and remove the third arg, `"bad :%v"` actually makes the error message worse because it doesn't tell you what the expected value is.